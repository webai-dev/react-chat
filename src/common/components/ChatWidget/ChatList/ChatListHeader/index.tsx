import React, { useState, useContext } from "react";
import { makeStyles } from "@material-ui/styles";
import Badge from "@material-ui/core/Badge";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import FilterListIcon from "@material-ui/icons/FilterList";
import { Company } from "../../../../../types/types";

import { ChatRoomsContext } from "../../../../hooks/useChatRooms";
import { CompanyContext } from "../../../../hooks/useCompanies";
import { RoomListsContext } from "../../../../hooks/useRoomLists";

const useStyles = makeStyles({
  root: {
    boxSizing: "border-box",
    display: "flex",
    alignItems: "center",
    height: "60px",
    padding: "10px 14px",
    backgroundColor: "#3870A0",
    color: "#ffffff",
  },
  badge: {
    backgroundColor: "#ffffff",
    color: "#665678",
    right: -10,
  },
  badgeAll: {
    backgroundColor: "#ffffff",
    color: "#665678",
    right: 12,
    top: -6,
  },
  companyItem: {
    padding: "12px 0 12px 38px",
  },
  dropdownContainer: {
    background: "#3870A0",
    width: "250px",
    color: "#fff",
    padding: 0,
    position: "absolute",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-around",
    margin: 0,
    top: 60,
    left: 0,
    cursor: "pointer",
    zIndex: 1000,
  },
});

interface ChatListHeaderProps {}

/**
 * ChatListHeader component
 * @param {ChatListHeaderProps} props Properties
 * @description ChatListHeader component consists of dropdown for switching between companies and connections
 */
const ChatListHeader: React.FC<ChatListHeaderProps> = () => {
  // States
  const [showDropdown, setShowDropdown] = useState(false);
  const { companies, selectedCompany, setSelectedCompany } =
    useContext(CompanyContext);
  const { companyRoomLists } = useContext(RoomListsContext);
  const { addChatRooms, hideAllRoom } = useContext(ChatRoomsContext);

  // Styles
  const classes = useStyles();

  //Event Handlers
  const handleClickDropdownItem = (
    e: React.MouseEvent<HTMLSpanElement, MouseEvent>,
    company: Company
  ) => {
    e.preventDefault();
    e.stopPropagation();
    setShowDropdown(!showDropdown);
    if (selectedCompany && company.id !== selectedCompany?.id) {
      // localStorage.setItem("chatRooms", JSON.stringify([]));

      if (selectedCompany) {
        hideAllRoom(selectedCompany.id);
      }
      hideAllRoom(company.id);
      setSelectedCompany(company);
      
      if (company) {
        localStorage.setItem("selectedCompany", JSON.stringify(company));
      }
      const item = localStorage.getItem("chatRooms");
      if (item) {
        let roomsList = JSON.parse(item);

        if (company && company.id) {
          let companyRooms = companyRoomLists.find(
            (c) => c.companyId === company.id
          );
          let roomIds = roomsList[company.id.toString()];
          if (companyRooms) {
            let rooms = companyRooms.rooms.map((r) => r.id);
            if (
              roomsList[company.id.toString()] &&
              roomsList[company.id.toString()].filter(
                (r: number[]) => r && rooms.includes(r[0])
              )
            ) {
              roomIds = roomsList[company.id.toString()].filter(
                (r: number[]) => r && rooms.includes(r[0])
              );
            }
          }

          addChatRooms(roomIds ? roomIds : [], company.id);
        } else if (company && company.id === null) {
          let companyRooms = companyRoomLists.find(
            (c) => c.companyId === company.id
          );
          let roomIds = roomsList["-1"];
          if (companyRooms) {
            let rooms = companyRooms.rooms.map((r) => r.id);
            if (
              roomsList["-1"] &&
              roomsList["-1"].filter((r: number[]) => r && rooms.includes(r[0]))
            ) {
              roomIds = roomsList["-1"].filter(
                (r: number[]) => r && rooms.includes(r[0])
              );
            }
          }

          addChatRooms(roomIds ? roomIds : [], company.id);
        }
      }
    }
  };

  const handleClickCompany = (
    e: React.MouseEvent<HTMLSpanElement, MouseEvent>
  ) => {
    e.preventDefault();
    e.stopPropagation();
    setShowDropdown(!showDropdown);
  };

  const totalUnreadCount =
    companies && companies.length > 1
      ? companies
          .map((company) => company.unreadCount)
          .reduce((a, b) => a + b, 0)
      : companies && companies[0]
      ? companies[0].unreadCount
      : 0;

  return (
    <div 
      data-testid="chat-list-header" 
      className={classes.root} 
      onClick={(e) => {
        handleClickCompany(e);
      }}
    >
      <Badge
        classes={{ badge: classes.badgeAll }}
        badgeContent={totalUnreadCount}
        color="primary"
      >
        <FilterListIcon />
      </Badge>

      <Typography>
        {selectedCompany ? selectedCompany.name : ""}
      </Typography>
      {showDropdown && (
        <Container className={classes.dropdownContainer}>
          {companies && companies.length > 1
            ? companies.map((company, index) => (
                <Typography
                  className={classes.companyItem}
                  onClick={(e) => handleClickDropdownItem(e, company)}
                  key={index}
                >
                  <Badge
                    classes={{ badge: classes.badge }}
                    badgeContent={company.unreadCount}
                    color="primary"
                  >
                    {company.name}
                  </Badge>
                </Typography>
              ))
            : ""}
        </Container>
      )}
    </div>
  );
};

export default ChatListHeader;
