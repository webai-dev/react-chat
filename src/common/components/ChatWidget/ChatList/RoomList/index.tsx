import React, {
  useState,
  useRef,
  useCallback,
  useContext,
  useEffect,
} from "react";
import Box from "@material-ui/core/Box";
import { makeStyles } from "@material-ui/styles";
import CircularProgress from "@material-ui/core/CircularProgress";
import RoomListItem from "./RoomListItem";
import { RoomListsContext } from "../../../../hooks/useRoomLists";
import { CompanyContext } from "../../../../hooks/useCompanies";
import { getRooms } from "../../../../services/rooms";
import { RoomListResponse } from "../../../../../types/types";
export const useStyles = makeStyles({
  root: {
    flex: 1,
    overflow: "auto",
  },
  loaderContainer: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
});

interface RoomListProps {
  connectionStatus: number;
}

/**
 * RoomList component
 * @param {{ RoomListProps }} props Properties
 * @description RoomList component consists of a list of RoomListItem components
 */
const RoomsList: React.FC<RoomListProps> = ({ connectionStatus = 1 }) => {
  const classes = useStyles();
  const [selectedRoomId, setSelectedRoomId] = useState(-1);
  const { selectedCompany } = useContext(CompanyContext);
  const {
    activeRoomList,
    loadNextPage,
    loadingList,
    loadingNextPage,
    companyRoomLists,
    setLoadingList,
  } = useContext(RoomListsContext);
  const containerRef = useRef() as React.MutableRefObject<HTMLInputElement>;

  useEffect(() => {
    if (connectionStatus === 1 && selectedCompany) {
      if (!activeRoomList) {
        setLoadingList(true);
        getRooms(selectedCompany.id, 0).then(
          (data: RoomListResponse | null) => {
            if (data) {
              companyRoomLists.push({
                companyId: selectedCompany.id,
                rooms: data.data,
                page: data.page,
                size: data.size,
                searchTerm: null,
                hasMore: data.totalPages > data.page,
              });

              setLoadingList(false);
            }
          }
        );
      } else {
        let updatedRoomsList = companyRoomLists.find(
          (c) => c.companyId === selectedCompany.id
        );
        if (updatedRoomsList) {
          setLoadingList(true);
          getRooms(selectedCompany.id, 0).then(
            (data: RoomListResponse | null) => {
              if (data && updatedRoomsList) {
                updatedRoomsList.companyId = selectedCompany.id;
                updatedRoomsList.rooms = data.data;
                updatedRoomsList.page = data.page;
                updatedRoomsList.size = data.size;
                updatedRoomsList.searchTerm = null;
                updatedRoomsList.hasMore = data.totalPages > data.page;

                setLoadingList(false);
              }
            }
          );
        }
        companyRoomLists.map((company) => {
          return company.companyId === selectedCompany.id
            ? updatedRoomsList
            : company;
        });
      }

      // updateActiveRooms();
    }
    // eslint-disable-next-line
  }, [connectionStatus]);

  // Infinite scrolling code
  const observer = useRef<IntersectionObserver>();
  const lastRoomElementRef = useCallback(
    (node) => {
      if (loadingNextPage) return;
      if (observer.current) observer.current.disconnect();
      observer.current = new IntersectionObserver((entries) => {
        if (entries[0].isIntersecting && activeRoomList?.hasMore) {
          loadNextPage();
          containerRef.current.scrollTop = containerRef.current.scrollHeight;
        }
      });
      if (node) observer.current.observe(node);
    },
    [loadingNextPage, activeRoomList?.hasMore, loadNextPage]
  );

  return (
    <div data-testid="room-list" className={classes.root} ref={containerRef}>
      {loadingList ? (
        <Box
          display="flex"
          height={410}
          justifyContent="center"
          alignItems="center"
        >
          <CircularProgress />
        </Box>
      ) : (
        <span>
          {activeRoomList?.rooms &&
            activeRoomList?.rooms.map((room, index) => (
              <div
                key={index}
                ref={
                  activeRoomList?.rooms.length === index + 1
                    ? lastRoomElementRef
                    : null
                }
              >
                <RoomListItem
                  connectionStatus={connectionStatus}
                  room={room}
                  selectedRoomId={selectedRoomId}
                  onClick={() => setSelectedRoomId(room.id)}
                />
              </div>
            ))}

          {loadingNextPage && (
            <Box
              display="flex"
              flex={1}
              height={90}
              justifyContent="center"
              alignItems="center"
            >
              <CircularProgress />
            </Box>
          )}
        </span>
      )}
    </div>
  );
};

export default RoomsList;
