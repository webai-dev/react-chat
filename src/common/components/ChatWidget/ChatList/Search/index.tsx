import React, { useState, useContext, useEffect } from "react";
import { makeStyles } from "@material-ui/styles";
import SearchIcon from "@material-ui/icons/Search";

import { RoomListsContext } from "../../../../hooks/useRoomLists";
import { TeamListsContext } from "../../../../hooks/useTeamLists";

export const useStyles = makeStyles({
  root: {
    display: "flex",
    height: "80px",
    backgroundColor: "#fff",
    paddingLeft: "30px",
    alignItems: "center",
    color: "#BCBCCB",
    fontSize: "13px",
  },
  textField: {
    border: "none",
    backgroundColor: "transparent",
    color: "#4D4F5C",
    "&:focus": {
      border: "none",
      outline: "none",
    },
  },
});

interface SearchProps {
  isTeamList?: boolean | undefined
}

/**
 * Search component
 * @param {{ SearchProps }} props Properties
 * @description Search component consists of a search icon and a text field for search value
 */
const Search: React.FC<SearchProps> = ({isTeamList}) => {
  // Styles
  const classes = useStyles();
  const [term, setTerm] = useState("");
  const { activeRoomList, setRoomListSearchTerm } =
    useContext(RoomListsContext);
  const { activeTeamList, setTeamListSearchTerm } =
    useContext(TeamListsContext);
  useEffect(() => {
    if (activeRoomList || activeTeamList) {
      setTerm(activeRoomList?.searchTerm || "");
    }
  }, [activeRoomList, activeTeamList]);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTerm(e.target.value);
  };

  const handleEnter = (target: React.KeyboardEvent) => {
    if (target.key === "Enter") {
      isTeamList ? setTeamListSearchTerm(term) : setRoomListSearchTerm(term);
    }
  };

  return (
    <div data-testid="search" className={classes.root}>
      <SearchIcon />
      <input
        type="text"
        placeholder="Search name..."
        value={term}
        className={classes.textField}
        onChange={handleChange}
        onKeyPress={handleEnter}
      />
    </div>
  );
};

export default Search;
