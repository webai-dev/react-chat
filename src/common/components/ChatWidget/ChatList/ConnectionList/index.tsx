import React, { useContext } from "react";
import Accordion from "@material-ui/core/Accordion";
import AccordionSummary from "@material-ui/core/AccordionSummary";
import AccordionDetails from "@material-ui/core/AccordionDetails";
import Box from "@material-ui/core/Box";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import { makeStyles } from "@material-ui/styles";
import ConnectionListContent from "./ConnectionListContent";

import { CompanyContext } from "../../../../hooks/useCompanies";
import Search from "../Search";

export const useStyles = makeStyles({
  accordion: {
    borderRadius: "0 !important",
    display: "flex",
    flexDirection: "column-reverse",
    position: "absolute",
    bottom: 0,
    right: 0,
    boxShadow: "0px 3px 6px #00000029",
    width: 370,
    maxHeight: "100%",

    "&:before": {
      backgroundColor: "transparent",
    },

    "& .MuiCollapse-container": {
      overflow: "auto",
    },
  },
  accordionDetails: {
    padding: 0,
    height: 491,
  },
  accordionIcon: {
    position: "absolute",
    color: "#fff",
    right: 12,
  },
  accordionSummary: {
    display: "flex",
    flexDirection: "row-reverse",
    justifyContent: "center",
    padding: 0,
    minHeight: "initial !important",
    borderTop: "1px solid rgba(0, 0, 0, 0.2)",
  },
  accordionSummaryContent: {
    margin: "0 !important",
    flexGrow: 0,
  },
  root: {
    flex: 1,
    overflow: "auto",
  },
  loaderContainer: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
});

interface ConnectionListProps {
  connectionStatus: number;
}

/**
 * TeamList component
 * @param {{ TeamListProps }} props Properties
 * @description TeamList component consists of a list of TeamListItem components
 */
const ConnectionList: React.FC<ConnectionListProps> = ({ connectionStatus = 1 }) => {
  const classes = useStyles();
  const { selectedCompany } = useContext(CompanyContext);

  return (
    <Accordion data-testid="team-list" className={classes.accordion}>
      <AccordionSummary
        classes={{
          root: classes.accordionSummary,
          content: classes.accordionSummaryContent,
        }}
        expandIcon={<ArrowUpwardIcon />}
      >
        <Box
          display="flex"
          alignItems="center"
          justifyContent="center"
          height={30}
          style={{ cursor: "pointer" }}
        >
          <Box marginLeft="0.25rem">Connections</Box>
        </Box>
      </AccordionSummary>
      <AccordionDetails className={classes.accordionDetails}>
        <Box width="100%" overflow="auto" display="flex" flexDirection="column">
          <Search />
          <ConnectionListContent
            connectionStatus={connectionStatus}
            companyId={selectedCompany && selectedCompany.id}
          />
        </Box>
      </AccordionDetails>
    </Accordion>
  );
};

export default ConnectionList;
