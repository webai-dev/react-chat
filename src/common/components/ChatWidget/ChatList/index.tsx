import React, { useContext } from "react";
import { makeStyles } from "@material-ui/styles";
import Accordion from "@material-ui/core/Accordion";
import AccordionSummary from "@material-ui/core/AccordionSummary";
import AccordionDetails from "@material-ui/core/AccordionDetails";
import Box from "@material-ui/core/Box";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";

import ChatListHeader from "./ChatListHeader";
import Search from "./Search";
import RoomList from "./RoomList";
import TeamList from "./TeamList";
import ConnectionList from "./ConnectionList";

import { CompanyContext } from "../../../hooks/useCompanies";
import { useState } from "react";

const useStyles = makeStyles({
  accordion: {
    borderRadius: "0 !important",
    position: "absolute",
    bottom: 0,
    right: 0,
    boxShadow: "0px 3px 6px #00000029",
    width: 370,
  },
  accordionDetails: {
    height: 540,
    padding: 0,
  },
  accordionExpanded: {
    margin: "0 !important",
  },
  accordionSummary: {
    padding: 0,
    minHeight: "initial !important",
    backgroundColor: "#3870A0",
    justifyContent: "space-between",
    paddingRight: 10,
  },
  accordionSummaryContent: {
    margin: "0 !important",
  },
  expandIcon: {
    color: "#ffffff",
    transform: "rotate(180deg)",
  },
  loaderContainer: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
});

interface ChatListProps {
  connectionStatus?: number;
}

/**
 * ChatList component
 * @param {ChatListProps} props Properties
 * @description ChatList component consists of ChatListHeader, Search, and Rooms/Team List components
 */
const ChatList: React.FC<ChatListProps> = ({ connectionStatus = 1 }) => {
  // Styles
  const classes = useStyles();

  const { selectedCompany } = useContext(CompanyContext);
  let chatListExpanded = localStorage.getItem("chatListExpanded");
  const [listExpanded, setListExpanded] = useState(
    chatListExpanded ? JSON.parse(chatListExpanded) : false
  );

  return (
    <div data-testid="chat-list">
      <Accordion
        expanded={listExpanded}
        classes={{
          root: classes.accordion,
          expanded: classes.accordionExpanded,
        }}
        onChange={(event: React.ChangeEvent<{}>, expand: boolean) => {
          localStorage.setItem("chatListExpanded", JSON.stringify(expand));
          setListExpanded(expand);
        }}
      >
        <AccordionSummary
          classes={{
            root: classes.accordionSummary,
            content: classes.accordionSummaryContent,
          }}
          expandIcon={<ExpandMoreIcon className={classes.expandIcon} />}
        >
          <ChatListHeader />
        </AccordionSummary>

        <AccordionDetails className={classes.accordionDetails}>
          <Box
            width="100%"
            overflow="hidden"
            display="flex"
            flexDirection="column"
          >
            <Search />
            <Box
              height={410}
              overflow="auto"
              data-testid="team-list"
            >
              <RoomList connectionStatus={connectionStatus} />
            </Box>
            {selectedCompany && selectedCompany.name === "Connections" && (
              <ConnectionList connectionStatus={connectionStatus} />
            )}
            {selectedCompany && selectedCompany.name !== "Connections" && (
              <TeamList connectionStatus={connectionStatus} />
            )}
          </Box>
        </AccordionDetails>
      </Accordion>
    </div>
  );
};

export default ChatList;
