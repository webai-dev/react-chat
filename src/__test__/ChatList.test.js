import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ChatList from "../common/components/ChatWidget/ChatList";
import renderer from "react-test-renderer";

it("renders without crashing", async () => {
  const { getByTestId } = render(<ChatList />);
  const chatList = getByTestId("chat-list");
  expect(chatList).toBeTruthy();
});

it("matches snapshot", async () => {
  const tree = renderer.create(<ChatList />).toJSON();
  expect(tree).toMatchSnapshot();
});
