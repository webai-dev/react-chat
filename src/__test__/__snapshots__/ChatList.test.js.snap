// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="chat-list"
>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-accordion-19 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-accordionSummary-22"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content makeStyles-accordionSummaryContent-23"
      >
        <div
          className="makeStyles-root-26"
          data-testid="chat-list-header"
        >
          <span
            className="MuiBadge-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
              />
            </svg>
            <span
              className="MuiBadge-badge makeStyles-badgeAll-28 MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
            >
              0
            </span>
          </span>
          <p
            className="MuiTypography-root MuiTypography-body1"
            onClick={[Function]}
          >
            
          </p>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-expandIcon-24"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-accordionDetails-20"
            >
              <div
                className="MuiBox-root MuiBox-root-31"
              >
                <div
                  className="makeStyles-root-32"
                  data-testid="search"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                  <input
                    className="makeStyles-textField-33"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Search name..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="MuiBox-root MuiBox-root-34"
                  data-testid="team-list"
                >
                  <div
                    className="makeStyles-root-35"
                    data-testid="room-list"
                  >
                    <span />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
